### PR TITLE
Logcollector macOS ULS documentation

### LINKED ISSUE #3651: Changes in the Documentation are required given the new OSX log collection method
|Linked Issue|
|---|
| wazuh/wazuh#4572 |

Hi Team!

As described in Epic wazuh/wazuh#4572, we are working in order to adapt the way Wazuh collects the logs on MacOS systems, as they use their own way to store the logs (**[unified logging system](https://developer.apple.com/documentation/os/logging)**). This will require some modifications to the documentation in order to inform the user how to configure the collection of the data on these systems.

### PR DESCRIPTION
**Pull request by nmkoremblum**
|Closes|
|---|
| #3651 |

Hi Team!

This PR, which closes #3651, was created in order to add the macOS' ULS log collection documentation.

This will include changes in:
- Log data collection
- Localfile (ossec.conf)

Additionally, we added a dedicated page to explain how can this feature be configured in the "Log data collection" section of the User Manual.

## Checks
- [x] It compiles without warnings.
- [ ] Spelling and grammar. 
- [ ] Used impersonal speech. 
- [ ] Used uppercase only on nouns. (not used)
- [ ] Updated the `redirect.js` script if necessary (check [this guide](https://github.com/wazuh/wazuh-documentation/blob/master/NEW_RELEASE.md)).

Regards,

Mariano Koremblum

### CODE REVIEW

File: source/user-manual/capabilities/log-data-collection/how-it-works.rst [hunk 1]
```diff
@@ -60,6 +60,26 @@ Wazuh can monitor classic Windows event logs, as well as the newer Windows event
       <log_format>eventchannel</log_format>
     </localfile>
 
+
+MacOS ULS logs
+^^^^^^^^^^^^^^
+
+.. versionadded:: 4.3.0
+
+Wazuh can monitor logs from the macOS Unified Logging System.
```
Comment (juliancnn) [inline source/user-manual/capabilities/log-data-collection/how-it-works.rst:69]:
```suggestion
Wazuh can monitor logs from the MacOS Unified Logging System.
```

File: source/user-manual/capabilities/log-data-collection/how-to-collect-macoslogs.rst [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+.. versionadded:: 4.3.0
+
+MacOS unified logging system (ULS) events can be gathered and forwarded to the manager.
+
+The ULS centralizes the management and storage of the logs across all the system levels, from Apps to the Kernel itself, rather than writing the data to text-based log files. Because of this, it is necessary to use specific methods to retrieve logs from this particular system.
+
+To gather the logs, Wazuh uses the CLI `log` tool which provides an interesting interface for log collection in a filtered way, since all the ULS logs pass through this single system. These parameters let the user configure the ``level`` of messages to be gathered, filter by the log ``type`` and even use a very specific ``predicate`` to filter the logs given their more specific characteristics.
+
+
+Monitor the macOS ULS Logs with Wazuh
```
Comment (juliancnn) [inline source/user-manual/capabilities/log-data-collection/how-to-collect-macoslogs.rst:17]:
```suggestion
Monitor the MacOS ULS Logs with Wazuh
```

File: source/user-manual/capabilities/log-data-collection/how-to-collect-macoslogs.rst [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    </localfile>
+
+To filter the system logs, it is necessary, but not mandatory, to use the ``query`` label. This label allows setting different filtering aspects such as:
+
+- ``type`` : Limits the type of logs that are intended to be acquired (``activity`` ; ``log`` **and/or** ``trace``). Multiple values are allowed.
+- ``level`` : Indicates the level of verbosity. Includes events at, and below, the given level value (``default`` ; ``info`` **or** ``debug``). For more information about the levels, see `ULS Logs Levels`_.
+- ``<query>process == "sshd" OR message CONTAINS "invalid"</query>`` : The query is used directly as the ULS predicate, which is used to filter the logs. For more information about the predicates, see `ULS Predicates`_.
+
+.. warning::
+    Be sure to be as restrictive as possible when filtering the logs. MacOS ULS produces a lot of data that might result overwhelming and some logs of interest could be lost.
```
Comment (juliancnn) [inline source/user-manual/capabilities/log-data-collection/how-to-collect-macoslogs.rst:37]: Perhaps explain buffer flooding 
https://documentation.wazuh.com/current/user-manual/capabilities/antiflooding.html

File: source/user-manual/reference/ossec-conf/localfile.rst [hunk 4]
```diff
@@ -289,6 +327,13 @@ Set the format of the log to be read. **field is required**
 |                    |                    |                                                                                                  |
 |                    |                    | This can be used to monitor standard “Windows” event logs and "Application and Services" logs.   |
 +                    +--------------------+--------------------------------------------------------------------------------------------------+
+|                    | oslog              | Used for macOS ULS logs, returns the logs in syslog format.                                      |
```
Comment (juliancnn) [inline source/user-manual/reference/ossec-conf/localfile.rst:330]:
```suggestion
|                    | oslog              | Used for MacOS ULS logs, returns the logs in syslog format.                                      |
```

File: source/user-manual/capabilities/log-data-collection/how-it-works.rst (hunk 1 above)
Comment (nmkoremblum) [reply source/user-manual/capabilities/log-data-collection/how-it-works.rst:69]: macOS is written with lowercase "m"

**Review by juliancnn — APPROVED** [review]

**Review by JcabreraC — APPROVED** [review]

**Review by juliancnn — APPROVED** [review]

**Review by JcabreraC — APPROVED** [review]
LGTM !

**Review by davidfmiranda — APPROVED** [review]
LGTM

**Pull request merged**